clarity/contracts/arkadiko-stake-pool-diko-xusd-v1-1.clar

```diff
@@ -219,24 +222,25 @@
 
 
 ;; Increase cumm reward per stake and save
-(define-public (increase-cumm-reward-per-stake)
+(define-public (increase-cumm-reward-per-stake (registry-trait <stake-registry-trait>))
   (let (
     ;; Calculate new cumm reward per stake
-    (new-cumm-reward-per-stake (calculate-cumm-reward-per-stake))
-    (last-block-height (get-last-block-height))
+    (new-cumm-reward-per-stake (unwrap-panic (calculate-cumm-reward-per-stake registry-trait)))
+    (last-block-height (get-last-block-height registry-trait))
   )
+    (asserts! (is-eq (contract-of registry-trait) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "stake-registry"))) ERR-WRONG-REGISTRY)
     (var-set cumm-reward-per-stake new-cumm-reward-per-stake)
     (var-set last-reward-increase-block last-block-height)
     (ok new-cumm-reward-per-stake)
   )
 )
 
 ;; Calculate current cumm reward per stake
-(define-read-only (calculate-cumm-reward-per-stake)
+(define-public (calculate-cumm-reward-per-stake (registry-trait <stake-registry-trait>))
```

Comment:
Q: shouldn't we check on correct stake registry here as well? A user can just call this with a random trait?

Reply:
A: the method does not update any variables, so not critical to check. Can add it.

Reply:
Ok - no worries then. No need to add it. Just additional read cycles then.